show/main.py

```diff
@@ -888,23 +892,213 @@ def get_bgp_peer():
+                    # MPLS labels are stored as an array (list) in json if present. Need to print through each one in list
+                    str_2_print += ", label "
+                    for k in range(0, len(info[i]['nexthops'][j]['labels'])):
+                        # MPLS labels that has value 15 or lower has special interpretation
+                        if info[i]['nexthops'][j]['labels'][k] > 15:
+                            label_string = str(info[i]['nexthops'][j]['labels'][k])
+                        else:
+                            label_string = mpls_label_code[info[i]['nexthops'][j]['labels'][k]]
+                        if k == 0:
+                            str_2_print += label_string
+                        else:
+                            str_2_print += "/" + label_string
+                # add uptime at the end of the string
+                str_2_print += " " + info[i]['uptime']
+                # print out this string
+                print str_2_print
+
+def print_show_ip_route_hdr():
+    # This prints out the show ip route header based on FRR 7.2 version.
+    # Please note that if we moved to future versions, we may heva to make changes to this
+    print("Codes: K - kernel route, C - connected, S - static, R - RIP,")
+    print("       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,")
+    print("       T - Table, v - VNC, V - VNC-Direct, A - Babel, D - SHARP,")
+    print("       F - PBR, f - OpenFabric,")
+    print("       > - selected route, * - FIB route, q - queued route, r - rejected route\n")
+
 #
 # 'route' subcommand ("show ip route")
 #
 
 @ip.command()
 @click.argument('args', metavar='[IPADDRESS] [vrf <vrf_name>] [...]', nargs=-1, required=False)
+@multi_asic_util.multi_asic_click_options
 @click.option('--verbose', is_flag=True, help="Enable verbose output")
-def route(args, verbose):
+def route(args, namespace, display, verbose):
     """Show IP (IPv4) routing table"""
-    cmd = 'sudo vtysh -c "show ip route'
+    is_multi_asic = multi_asic.is_multi_asic()
+    if is_multi_asic:
+        # handling multi-ASIC by gathering the output from specified/all name space into a dictionary via 
+        # jason option and then filter out the json entries (by removing those next Hop that are 
+        # back-end ASIC if display is for front-end only). If the entry itself has no more next Hop after filtering, 
+        # then skip over that particular route entry.  Once completed, if the user chose "json" option, 
+        # then just print out the dictionary in Json format accordingly. But if no "json" option specified,
+        # then print out the header and the decoded entry representation for each route accordingly.
+        # This code is based on FRR 7.2 branch. If we moved to a new version we may need to change here as well
+        asic_cnt = multi_asic.get_num_asics()
+        arg_strg = ""
+        found_json = 0
+        back_end_intf_str = "PortChannel4"
```

Comment:
Can we use this function to filter display of internal routes [skip_display](https://github.com/Azure/sonic-utilities/blob/58c29616f510ebd94122db4490523174e1256943/utilities_common/multi_asic.py#L33) instead of hardcording

Reply:
Agreed. Have Changed to use skip_display to detect skip required.